### PR TITLE
add hs with AMIP targeted resolution to longrun

### PR DESCRIPTION
**Pull request by jiahe23**
# PULL REQUEST

## Purpose and Content
This PR adds high resolution hs simulations to longrun buildkite.
It sets up mpi with 128 tasks.

## Benefits and Risks
(State concisely the benefits to be derived from and the risks associated with this PR)

## Linked Issues
(Provide references to any link issues. Use closes #issuenum to automatically close an open issue)
- Fixes #
- Closes #

## PR Checklist
- [ ] This PR has a corresponding issue OR is linked to an SDI.
- [ ] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [ ] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [ ] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [ ] I linted my code on my local machine prior to submission OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code used in an integration test OR N/A.
- [ ] All tests ran successfully on my local machine OR N/A.
- [ ] All classes, modules, and function contain docstrings OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Review by charleskawczynski — COMMENTED** [review]
I think it’s a good idea to add this in general, but there are some issues with the flags used. I’ll try to confirm the correct format today

**Comment by jiahe23** [comment]
> I think it’s a good idea to add this in general, but there are some issues with the flags used. I’ll try to confirm the correct format today

Thanks! That's very helpful.

**Comment by charleskawczynski** [comment]
Maybe we can discuss this during the amip meeting today. The terminology is a bit confusing and we're running different parallelization models in different ways. It might be nice to document what works.

**Comment by jiahe23** [comment]
> Maybe we can discuss this during the amip meeting today. The terminology is a bit confusing and we're running different parallelization models in different ways. It might be nice to document what works.

Yes, that will be helpful. Here I'm following comments in #660 which says 
```
  agents: # works
    slurm_ntasks: 8
    slurm_ntasks_per_node: 8
```

**Comment by charleskawczynski** [comment]
> Yes, that will be helpful. Here I'm following comments in #660 which says...

Yeah, but this is for running jobs with multiple threads, not multiple processors.

File: .buildkite/longruns/pipeline.yml [hunk 1]
```diff
@@ -116,3 +116,22 @@ steps:
...
+        agents:
+          slurm_ntasks: 128
+
+      - label: ":computer: held suarez (ρe_tot) equilmoist high resolution"
+        command: "mpiexec julia --project=examples examples/hybrid/driver.jl --forcing held_suarez --moist equil --vert_diff true --microphy 0M --z_elem 45 --dz_bottom 30 --h_elem 12 --kappa_4 7e15 --dt 50secs --t_end 150days --dt_save_to_disk 10days --job_id longrun_hs_rhoe_equil_highres"
+        artifact_paths: "longrun_hs_rhoe_equil_highres/*"
+        env:
+          CLIMACORE_DISTRIBUTED: "MPI"
+        agents:
+          slurm_ntasks: 256
```
Comment (charleskawczynski) [inline .buildkite/longruns/pipeline.yml:137]: Why do we have both moist and dry?

File: .buildkite/longruns/pipeline.yml [hunk 2]
```diff
@@ -116,3 +116,22 @@ steps:
         agents:
           slurm_cpus_per_task: 8
 
+  - group: "Targeted AMIP long runs"
```
Comment (jiahe23) [inline .buildkite/longruns/pipeline.yml:119]:
```suggestion
  - group: "Targeted resolution AMIP long runs"
```

**Pull request merged**